extensions/common/json-ld/src/main/java/org/eclipse/edc/jsonld/TitaniumJsonLd.java

```diff
@@ -191,14 +193,20 @@ private static class CachedDocumentLoader implements DocumentLoader {
         @Override
         public Document loadDocument(URI url, DocumentLoaderOptions options) throws JsonLdError {
             var uri = Optional.of(url.toString())
-                    .map(cache::get)
+                    .map(uriCache::get)
                     .orElse(url);
 
-            return loader.loadDocument(uri, options);
+            return Optional.of(documentCache.get(uri))
+                    .orElse(loader.loadDocument(uri, options));
         }
 
         public void register(String contextUrl, URI uri) {
-            cache.put(contextUrl, uri);
+            uriCache.put(contextUrl, uri);
+            try {
+                documentCache.put(uri, loader.loadDocument(uri, new DocumentLoaderOptions()));
+            } catch (JsonLdError e) {
+                throw new EdcException(e);
```

Comment:
maybe we could add an exception message like "Error in caching json ld context %s from URI %s"